x-pack/filebeat/input/httpjson/input_test.go

```diff
@@ -25,12 +26,14 @@ import (
 )
 
 func TestInput(t *testing.T) {
+	tempDirectory := t.TempDir()
```

Comment:
I would rather see each sub-test use its own unique temp dir to ensure test isolation. But with only one test using this directory it's only a minor concern.

Reply:
I had to change the structure to achieve this. To inject `tempDir` into the config which is currently a `Map` I felt difficult with current setup. As you said for one test using it , this should be a big change for the value it gives.